src/charm.py

```diff
@@ -79,79 +80,182 @@ def _config_ca_common_name(self) -> Optional[str]:
         return self.model.config.get("ca-common-name", None)
 
     @property
-    def _certificates_are_valid(self) -> bool:
+    def _config_certificates_are_valid(self) -> bool:
         """Returns whether user provided certificates are valid.
 
         Returns:
             bool: True/False
         """
         try:
-            certificate_bytes = base64.b64decode(self.model.config.get("certificate"))  # type: ignore[arg-type]  # noqa: E501
-            ca_certificate_bytes = base64.b64decode(self.model.config.get("ca-certificate"))  # type: ignore[arg-type]  # noqa: E501
-            ca_chain_bytes = base64.b64decode(self.model.config.get("ca-chain"))  # type: ignore[arg-type]  # noqa: E501
-        except binascii.Error:
+            certificate_bytes = base64.b64decode(self.model.config.get("certificate", None))  # type: ignore[arg-type]  # noqa: E501
+            ca_certificate_bytes = base64.b64decode(self.model.config.get("ca-certificate", None))  # type: ignore[arg-type]  # noqa: E501
+        except (binascii.Error, TypeError):
             return False
-        try:
-            assert certificate_is_valid(certificate_bytes)
-            assert certificate_is_valid(ca_certificate_bytes)
-            assert certificate_is_valid(ca_chain_bytes)
-            return True
-        except AssertionError:
+        if not certificate_is_valid(certificate_bytes):
+            logger.error("Config `certificate` is not valid")
             return False
+        if not certificate_is_valid(ca_certificate_bytes):
+            logger.error("Config `ca-certificate` is not valid")
+            return False
+        config_ca_chain = self.model.config.get("ca-chain", None)
+        if config_ca_chain:
+            try:
+                ca_chain_list = json.loads(
+                    self.model.config.get("ca-chain")  # type: ignore[arg-type]
+                )
+                ca_chain_bytes_list = [base64.b64decode(item) for item in ca_chain_list]
+            except (binascii.Error, TypeError):
+                return False
```

Comment:
This code is currently broken. I am not sure how to fix this, because we want to get a list for the CA chain, but `charmcraft` does not support configurations elements of that type. Supported types are documented here: https://juju.is/docs/sdk/config

We could try and work around the issue by "faking" a list with separators inside the string, but it would be ugly, and hard to use. 

The other option I could see is that there might be a way in the cryptography library to load a whole chain from a single string, as certificate chains are usually stored in a single `PEM` file.

Reply:
I have pushed a small commit to pass the `ca-chain` in the tests as a JSON list. This works, but is probably not great for users. It at least fixes the tests for now and might unblock us.

Reply:
I moved to using ca chains as a single pem file that contains all the certs.

Reply:
Great, that works for me.